### PR TITLE
[BEAM-9298] Drop support for Flink 1.7

### PR DESCRIPTION
**Pull request by sunjincheng121**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: website/src/documentation/runners/flink.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ To find out which version of Flink is compatible with Beam please see the table
   <th>Artifact Id</th>
 </tr>
 <tr>
```
Comment (tweise) [inline website/src/documentation/runners/flink.md]: We need a new row in this table that specifies the version support from 2.20 (or whatever version we release this change with) going forward.

Comment (sunjincheng121) [reply website/src/documentation/runners/flink.md]: Sure，will update the PR after https://github.com/apache/beam/pull/10945 merged. :)

**Comment by sunjincheng121** [comment]
The PR should be update, So, please review the changes after #10945 merged. :) Thank you!

**Comment by mxm** [comment]
@sunjincheng121 Could you update the PR so we can get this in?

**Comment by sunjincheng121** [comment]
> @sunjincheng121 Could you update the PR so we can get this in?

Sure, Thanks for the reminder @mxm , I would like to update this PR soon. :)

**Comment by sunjincheng121** [comment]
Run CommunityMetrics PreCommit

**Comment by sunjincheng121** [comment]
Run Python PreCommit

File: website/src/documentation/runtime/environments.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ python -m apache_beam.examples.wordcount \
 
 ```
 # Start a Flink job server on localhost:8099
-./gradlew :runners:flink:1.7:job-server:runShadow
+./gradlew :runners:flink:1.8:job-server:runShadow
```
Comment (mxm) [inline website/src/documentation/runtime/environments.md]: Perhaps coud be 1.10 now.

File: website/src/documentation/runners/flink.md (hunk 1 above)
Comment (mxm) [reply website/src/documentation/runners/flink.md]: This has been addressed in #10945 and updated in this PR. 2.21.0 will be the first version without 1.7 and with 1.10 support.

File: website/src/documentation/runners/flink.md [hunk 3]
```diff
@@ -103,7 +103,7 @@ To find out which version of Flink is compatible with Beam please see the table
...
-  <td rowspan="3">2.20.0</td>
+  <td rowspan="3">2.21.0</td>
```
Comment (mxm) [inline website/src/documentation/runners/flink.md]: This should probably be `>2.21.0` but not important for merging this PR.

**Comment by sunjincheng121** [comment]
Thank you @mxm !

**Pull request merged**